code/lib/core-common/src/utils/paths.ts

```diff
@@ -1,31 +1,35 @@
 import path from 'path';
 import findUp from 'find-up';
 
-export const getProjectRoot = () => {
-  let result;
-  try {
-    const found = findUp.sync('.git', { type: 'directory' });
-    if (found) {
-      result = result || path.join(found, '..');
-    }
-  } catch (e) {
-    //
-  }
-  try {
-    const found = findUp.sync('.svn', { type: 'directory' });
-    if (found) {
-      result = result || path.join(found, '..');
-    }
-  } catch (e) {
-    //
+// Try and figure out what the root folder of the project is.  This is used to generate some configuration.
+export const getProjectRoot = (): string => {
+  // Allow manual override in cases where the auto-detect doesn't work
+  if (process.env.STORYBOOK_PROJECT_ROOT) {
+    return process.env.STORYBOOK_PROJECT_ROOT;
   }
-  try {
-    result = result || __dirname.split('node_modules')[0];
-  } catch (e) {
-    //
+  // Assume the innermost git/svn/hg folder is the project root
+  const foundVcsMetaDir = ['.git', '.svn', '.hg']
+    .map((dir) => findUp.sync(dir, { type: 'directory' }))
```

Comment:
This is wasteful, especially considering all these are blocking the main thread sync calls.

The old code before only tried .svn if .git failed.
The new code will try all 3, even if the first one succeeds.

Reply:
True